javascript/spec/test.js

```diff
@@ -0,0 +1,66 @@
+import { expect } from 'chai'
+import oo from '../object_oriented.js'
+import functional from '../functional.js'
+import imperative from '../imperative.js'
+
+const test = (input, output) => {
+  const paradigms = [
+    ['Object Oriented', oo]
+    , ['Functional', functional]
+    , ['Imperative', imperative]
+  ]
+
+  for (let i in paradigms) {
+    const [name, fn] = paradigms[i]
+    it(name, () => {
+      expect(fn(input)).to.equal(output)
+    })
+  }
+}
+
+const nest = (element) => {
```

Comment:
anyway, I'm thinking of pre-emptively giving myself much less work by refactoring the whole thing so it autogenerates tests for every tag and all I have to input is a configuration object of form { name: tag }